src/metrics/include/metrics/operation.hpp

```diff
@@ -92,15 +91,40 @@ struct OperationEvent final {
     OutcomeType outcome = OutcomeType::kUnknown;  // corresponds to the 'outcome' field in Cedar
 };
 
+class OperationThresholdExceededException : public virtual boost::exception,
+                                            public virtual std::exception {};
 
 template <typename ClockSource>
 class OperationImpl final : private boost::noncopyable {
+private:  // Data members.
+    struct OperationCount {
+        int64_t failed = 0;
+        int64_t total = 0;
+
+        constexpr double_t failedPercentage() const {
+            return static_cast<double_t>(failed) / total * 100;
+        };
+    };
+
 public:
     using time_point = typename ClockSource::time_point;
     using EventSeries = TimeSeries<ClockSource, OperationEvent<ClockSource>>;
 
-    OperationImpl(std::string actorName, std::string opName)
-        : _actorName(std::move(actorName)), _opName(std::move(opName)) {}
+    struct OperationThreshold {
+        std::chrono::nanoseconds duration;
+        double_t failedPercentageThreshold;
+        OperationCount counter;
```

Comment:
1. Rename `duration` to `maxDuration`?
1. Rename counter to `occurrences` or `seenEvents`?
1. Rename `failedPercentageThreshold` to `percentAllowedToExceed` or something?

Reply:
+1 for `maxDuration`. 

I like `counter` or `opCounter` better. 

+1 on `percentAllowedToExceed` or `maxPercentAllowedToExceed`